gnomad_qc/v5/sample_qc/relatedness.py

```diff
@@ -101,11 +111,198 @@ def convert_cuking_output_to_ht(cuking_output_path: str) -> hl.Table:
+        - 'released_gnomad_exomes_duplicate': Whether the sample pair contains a released
+          gnomAD exomes duplicate.
+        - 'released_gnomad_genomes_duplicate': Whether the sample pair contains a released
+          gnomAD genomes duplicate.
+
+    :param ht: Input relatedness Table.
+    :param meta_ht: Input metadata Table. Used to add v4 overlap annotations.
+    :param relatedness_args: Dictionary of arguments to be passed to
+        `get_slope_int_relationship_expr`.
+    :return: Finalized relatedness Table
+    """
+    y_expr = ht.ibs0 / ht.ibs2
+    ibd1_expr = None
+
+    ht = ht.annotate(
+        ibs0_2=ht.ibs0 / ht.ibs2,
+        relationship=get_slope_int_relationship_expr(
+            kin_expr=ht.kin,
+            y_expr=y_expr,
+            **relatedness_args,
+            ibd1_expr=ibd1_expr,
+        ),
+    )
+    # The function get_slope_int_relationship_expr expects parent_child_max_y for
+    # flexibility but revert back to parent_child_max_ibs0_over_ibs2 for global clarity
+    relatedness_args["parent_child_max_ibs0_over_ibs2"] = relatedness_args.pop(
+        "parent_child_max_y"
+    )
+    ht = ht.annotate_globals(
+        relationship_inference_method="cuking",
+        relationship_cutoffs=hl.struct(**relatedness_args),
+    )
+
+    # Mark AoU samples as duplicates of gnomAD samples
+    ht = ht.key_by(
+        i=ht.i.annotate(
+            data_type=meta_ht[ht.i.s].data_type,
+            project=meta_ht[ht.i.s].project,
+            release=meta_ht[ht.i.s].release,
+        ),
+        j=ht.j.annotate(
+            data_type=meta_ht[ht.j.s].data_type,
+            project=meta_ht[ht.j.s].project,
+            release=meta_ht[ht.j.s].release,
+        ),
+    )
+
+    is_aou_gnomad_duplicate = (ht.relationship == DUPLICATE_OR_TWINS) & (
+        hl.set({ht.i.project, ht.j.project}) == hl.set(["aou", "gnomad"])
+    )
+
+    # Get the gnomAD sample's data type and release status (if one exists in the pair)
+    gnomad_data_type = hl.if_else(
+        ht.i.project == "gnomad",
+        ht.i.data_type,
+        hl.if_else(ht.j.project == "gnomad", ht.j.data_type, hl.missing(hl.tstr)),
+    )
+
+    gnomad_release_status = hl.if_else(
+        ht.i.project == "gnomad",
+        ht.i.release,
+        hl.if_else(ht.j.project == "gnomad", ht.j.release, hl.missing(hl.tbool)),
+    )
+
+    ht = ht.annotate(
+        is_aou_gnomad_duplicate=is_aou_gnomad_duplicate,
+        gnomad_data_type=gnomad_data_type,
+        gnomad_exomes_aou_duplicate=is_aou_gnomad_duplicate
+        & (gnomad_data_type == "exomes"),
+        gnomad_genomes_aou_duplicate=is_aou_gnomad_duplicate
+        & (gnomad_data_type == "genomes"),
+        released_gnomad_exomes_aou_duplicate=is_aou_gnomad_duplicate
+        & (gnomad_data_type == "exomes")
+        & gnomad_release_status,
+        released_gnomad_genomes_duplicate=is_aou_gnomad_duplicate
+        & (gnomad_data_type == "genomes")
+        & gnomad_release_status,
+    )
```

Comment:
This is the annotation block where I suspect the logic is incorrect either here or in the expression declaration above